### PR TITLE
Add build infrastructure for org.eclipse.emfcloud

### PR DESCRIPTION
**Pull request by sgraband**
- Refactored java package names to org.eclipse.emfcloud
- Added emfjson-p2-build for building a p2-update-site
- Added maven profiles for building and deploying to sonatype
- Changed groupId of maven and p2 to org.eclipse.emfcloud

Co-authored-by: Tobias Ortmayr <tortmayr@eclipsesource.com>

### CODE REVIEW

**Review by planger — COMMENTED** [review]
Looks great overall! Thanks a lot @sgraband!
I was just thinking whether we should mark in the version ID of the published artifacts that it is a nightly? Except for the URL, it actually has a release version ID.
I know we do the same for GLSP, but I was just wondering whether we should change that. EMF for instance mark nightly releases with N....
But I guess this is just cosmetics. WDYT?

File: pom.xml [hunk 1]
```diff
@@ -1,344 +1,415 @@
...
-
-    </build>
+	<parent>
+		<groupId>org.sonatype.oss</groupId>
+		<artifactId>oss-parent</artifactId>
+		<version>7</version>
+	</parent>
+
+	<licenses>
+		<license>
```
Comment (tortmayr) [inline pom.xml:18]: Wrong license information. Should be:
```xml
	<licenses>
		<license>
			<name>Eclipse Public License - v2.0</name>
			<url>http://www.eclipse.org/legal/epl-v20.html</url>
			<distribution>repo</distribution>
		</license>
		<license>
			<name>MIT License</name>
			<url>https://opensource.org/licenses/MIT</url>
			<distribution>repo</distribution>
		</license>
	</licenses>
```

File: pom.xml [hunk 2]
```diff
@@ -1,344 +1,415 @@
...
+	</licenses>
+
+	<issueManagement>
+		<system>GitHub Issues</system>
+		<url>https://github.com/eclipse-emfcloud/emfjson-jackson/issues</url>
+	</issueManagement>
+
+	<developers>
+		<developer>
+			<name>Eclipse emf.cloud Project</name>
```
Comment (tortmayr) [inline pom.xml:31]:
```suggestion
			<name>Eclipse EMF.cloud Project</name>
```

File: pom.xml [hunk 3]
```diff
@@ -1,344 +1,415 @@
...
+			<email>emfcloud-dev@eclipse.org</email>
+			<organization>Eclipse emf.cloud</organization>
```
Comment (tortmayr) [inline pom.xml:33]:
```suggestion
			<organization>Eclipse EMF.cloud</organization>
```

File: pom.xml [hunk 4]
```diff
@@ -1,344 +1,415 @@
...
+		</license>
+	</licenses>
+
+	<issueManagement>
+		<system>GitHub Issues</system>
+		<url>https://github.com/eclipse-emfcloud/emfjson-jackson/issues</url>
+	</issueManagement>
+
+	<developers>
+		<developer>
```
Comment (tortmayr) [inline pom.xml:30]: We should keep the developers info of the original maintainer and add the EMF.cloud standin developer as additional developer.

**Review by tortmayr — COMMENTED** [review]
Looks good to me. I have a couple of comments/suggestions regarding the metadata in the pom.xml

**Comment by sgraband** [comment]
I updated the metadata in the pom.xml. Regarding the indication of a nightly build i think the timestamp in the version number is good enough of a indicator, since we will not have that on proper releases, right? How do you think about that?

**Comment by tortmayr** [comment]
Good enough for me. @planger Should we merge this?

File: pom.xml [hunk 5]
```diff
@@ -1,344 +1,426 @@
...
-                        <artifactId>org.eclipse.emf.ecore.xcore.lib</artifactId>
-                        <version>${ecore-xcore-lib-version}</version>
-                    </dependency>
-                </dependencies>
-            </plugin>
-
-        </plugins>
-
-    </build>
+		<developer>
```
Comment (tortmayr) [inline pom.xml:41]: Formmating seems to be off.

**Comment by planger** [comment]
Yes, I don't see a reason for waiting with merging this change. Do you?
Thanks Simon for the work!

**Pull request merged**